postgres/postgres.go

```diff
@@ -465,26 +493,71 @@ func (ds Datastore) UpdateFilter(isAuthenticated bool, filter *models.Filter) er
 		return err
 	}
 
-	updateStatement, err := updateStatement(filter)
-	if err != nil {
-		return err
+	rowsAffected := int64(0)
+
+	if filter.State != "" {
+		results, err := tx.Stmt(ds.updateFilterState).Exec(filter.FilterID, filter.State)
+		if err != nil {
+			if err = tx.Rollback(); err != nil {
+				return err
+			}
+			return err
+		}
+
+		rowsAffected, err = results.RowsAffected()
+		if err != nil {
+			if err = tx.Rollback(); err != nil {
+				return err
+			}
+			return err
+		}
 	}
 
-	stmt, err := tx.Prepare(updateStatement)
-	if err != nil {
-		return err
+	if !isAuthenticated && rowsAffected == 0 {
+		return errors.New("Bad request")
 	}
 
-	_, err = stmt.Exec(filter.FilterID)
-	if err != nil {
-		return err
+	csv := filter.Downloads.CSV
+	json := filter.Downloads.JSON
+	xls := filter.Downloads.XLS
+
+	if isAuthenticated {
```

Comment:
You may not need to check isAuthenticated here. If the request is not authenticated the method would have already returned from the previous check?